base/range.jl

```diff
@@ -801,3 +801,5 @@ end
 
 in{T<:Integer}(x, r::Range{T}) = isinteger(x) && !isempty(r) && x>=minimum(r) && x<=maximum(r) && (mod(convert(T,x),step(r))-mod(first(r),step(r)) == 0)
 in(x::Char, r::Range{Char}) = !isempty(r) && x >= minimum(r) && x <= maximum(r) && (mod(Int(x) - Int(first(r)), step(r)) == 0)
+
+unique{T}(x::Range{T}) = step(x) == zero(T) ? x[1:1] : x
```

Comment:
Oops, didn't see this before now.

I don't quite understand your code, here. Did you mean `!=` instead of `==`, or did you mean `return x[1]`. But even so
```
julia> x = linspace(0.0, nextfloat(0.0), 3)
linspace(0.0,5.0e-324,3)
julia> x[1] != x[2]
false
julia> x[1] != x[end]
true
```
The problem here is that I'm asking for more elements than there are actually floats in the interval. One way of checking for this is
```
last(s) >= nextfloat(first(s), length(s) - 1)
```
That should at least _rule out_ the problem.

But I'm not sure what is the best way to construct a minimal unique version for all float ranges. What happens when you cross from one `eps` size to another, for example?